pkg/sign/signer.go

```diff
@@ -15,99 +15,74 @@
 package sign
 
 import (
-	"crypto"
-	"crypto/rand"
-	"crypto/sha256"
-	_ "crypto/sha512" // if user chooses SHA2-384 or SHA2-512 for hash
-	"crypto/x509"
-	"encoding/base64"
-	"errors"
-
 	protobundle "github.com/sigstore/protobuf-specs/gen/pb-go/bundle/v1"
 	protocommon "github.com/sigstore/protobuf-specs/gen/pb-go/common/v1"
 )
 
-type Signer interface {
-	Sign(data []byte) (*protobundle.Bundle, error)
-}
+const bundleV03MediaType = "application/vnd.dev.sigstore.bundle.v0.3+json"
 
-// TODO: implement
-// type Fulcio struct {
-//	baseURL       string
-//	identityToken string
-// }
+func Bundle(content Content, keypair Keypair, fulcio *Fulcio, idToken string, timestampAuthority *TimestampAuthority) (*protobundle.Bundle, error) {
+	// See if keypair was provided
+	if keypair == nil {
```

Comment:
I'm not sure about this. For critical things such as signing I would suggest we require all parameters explicitly.

Reply:
Yeah, I was thinking that in the frequent Fulcio case, people would just want an ephemeral keypair.

But it's very easy to instantiate an ephemeral keypair with `pkg/sign/keys.go`, and I can see the point about this being better if its explicit, so I removed this.